SPHINXsys/src/shared/simbody_sphinxsys/xml_engine.h

```diff
@@ -92,28 +96,30 @@ namespace SPH
 			SimTK::Xml::Attribute attr_(attrib_name, SimTK::String(array_));
 			ele_ite->setAttributeValue(attr_.getName(), attr_.getValue());
 		};
-		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const int &value)
-		{
-			SimTK::Xml::Attribute attr_(attrib_name, SimTK::String(value));
-			ele_ite->setAttributeValue(attr_.getName(), attr_.getValue());
-		};
 
-		/** Get the required attribute value of an element */
-		void getRequiredAttributeValue(SimTK::Xml::element_iterator &ele_ite_, const std::string &attrib_name, Real &value)
-		{
-			std::string value_in_string = ele_ite_->getRequiredAttributeValue(attrib_name);
-			value = SimTK::convertStringTo<Real>(value_in_string);
-		};
+		void setAttributeToElement(const SimTK::Xml::element_iterator &ele_ite, const std::string &attrib_name, const Mat2d &value);
+		void setAttributeToElement(const SimTK::Xml::element_iterator& ele_ite, const std::string& attrib_name, const Mat3d& value);
+
+		//----------------------------------------------------------------------
+		//	Get the required attribute value of an element.
+		//----------------------------------------------------------------------
+		void getRequiredAttributeValue(SimTK::Xml::element_iterator &ele_ite_, const std::string &attrib_name, Real &value);
+		void getRequiredAttributeValue(SimTK::Xml::element_iterator &ele_ite_, const std::string &attrib_name, int &value);		
+		
 		template <int DIMENSION>
-		void getRequiredAttributeValue(SimTK::Xml::element_iterator &ele_ite_, const std::string &attrib_name,
-									   Eigen::Matrix<Real, DIMENSION, 1> &value)
+		void getVectorAttributeValue(SimTK::Xml::element_iterator& ele_ite_, const std::string& attrib_name,
+			Eigen::Matrix<Real, DIMENSION, 1>& value)
 		{
 			std::string value_in_string = ele_ite_->getRequiredAttributeValue(attrib_name);
 			value = SimTKToEigen(SimTK::convertStringTo<SimTK::Vec<DIMENSION>>(value_in_string));
 		};
+
+		void getRequiredAttributeValue(SimTK::Xml::element_iterator &ele_ite_, const std::string &attrib_name, Vec2d &value);
+		void getRequiredAttributeValue(SimTK::Xml::element_iterator& ele_ite_, const std::string& attrib_name, Vec3d& value);
+
 		template <int DIMENSION>
-		void getRequiredAttributeValue(SimTK::Xml::element_iterator &ele_ite_, const std::string &attrib_name,
-									   Eigen::Matrix<Real, DIMENSION, DIMENSION> &value)
+		void getMatrixAttributeValue(SimTK::Xml::element_iterator& ele_ite_, const std::string& attrib_name,
+			Eigen::Matrix<Real, DIMENSION, DIMENSION>& value)
 		{
 			std::string value_in_string = ele_ite_->getRequiredAttributeValue(attrib_name);
 			SimTK::Array_<Real, int> array_;
```

Comment:
Why is this done only for matrices and not vectors ?

Both should be serializable uniformly according to [SimTK::String(const T&)](https://simtk.org/api_docs/simbody/api_docs33/Simbody/html/classSimTK_1_1String.html#ae1c3122a261b2a809607040facdd7c07) and [operator<< for SimTK::Mat](https://simtk.org/api_docs/simbody/api_docs33/Simbody/html/namespaceSimTK.html#abf8a923ce5ce85ac3fddbddd1c5df70d). This would reduce duplication and simplify the code to do the same for vectors and matrices.